plugins/BEdita/Core/src/Model/Action/CloneObjectAction.php

```diff
@@ -131,6 +132,18 @@ protected function setEntityField(array $schemaInfo, ObjectEntity $sourceEntity,
             return $attributes[$field];
         }
         $value = $sourceEntity->get($field);
+        if ($sourceEntity->getTable()->hasAssociation(Inflector::camelize($field))) {
+            $value = array_map(function ($data) {
+                $entityClass = get_class($data);
+                unset($data['id']);
+                unset($data['object_id']);
+
+                return new $entityClass($data->toArray());
+            }, $value);
+            $entity->set($field, $value);
+
+            return $value;
+        }
```

Comment:
:thinking: Since the associations could be of many types (`HasOne`, `BelongsTo`, `HasMany`, `BelongsToMany`) it doesn't always work.

To keep it simple we could refer to `HasMany` association for now doing something as

```php
$association = $sourceEntity->getTable()->associations()->getByProperty($field);
if ($association instanceof HasMany) {
    $targetTable = $association->getTarget();
    // foreignKey could be an array... but let's keep it simple
    $foreignKey = $targetTable->getForeignKey();
    $value = array_map(function ($data) {
        unset($data['id']);
        unset($data[$foreignKey]);

    return $targetTable->newEntity($data->toArray());
    }, $value);
    $entity->set($field, $value);

    return $value;
}
```